kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/service/impl/ExtractPaymentServiceImpl.java

```diff
@@ -0,0 +1,913 @@
+        this.paymentGroupHistoryDao = paymentGroupHistoryDao;
+    }
+
+    /**
+     * Sets the processDao attribute value.
+     *
+     * @param processDao The processDao to set.
+     */
+    @Override
+    public void setProcessDao(ProcessDao processDao) {
+        this.processDao = processDao;
+    }
+
+    /**
+     * Sets the paymentFileEmailService attribute value.
+     *
+     * @param paymentFileEmailService The paymentFileEmailService to set.
+     */
+    @Override
+    public void setPaymentFileEmailService(PdpEmailService paymentFileEmailService) {
+        this.paymentFileEmailService = paymentFileEmailService;
+    }
+
+    /**
+     * Sets the business object service
+     *
+     * @param businessObjectService
+     */
+    @Override
+    public void setBusinessObjectService(BusinessObjectService businessObjectService) {
+        this.businessObjectService = businessObjectService;
+    }
+
+    /**
+     * Gets the countryService attribute.
+     *
+     * @return Returns the countryService.
+     */
+    @Override
+    protected CountryService getCountryService() {
+        return countryService;
+    }
+
+    /**
+     * Sets the countryService attribute value.
+     *
+     * @param countryService The countryService to set.
+     */
+    @Override
+    public void setCountryService(CountryService countryService) {
+        this.countryService = countryService;
+    }
+
+    /**
+     * @see org.kuali.kfs.sys.batch.service.impl.InitiateDirectoryImpl#getRequiredDirectoryNames()
+     */
+    @SuppressWarnings("serial")
+	@Override
+    public List<String> getRequiredDirectoryNames() {
+        return new ArrayList<String>() {{add(directoryName); }};
+    }
+
+    @Override
+    public DataDictionaryService getDataDictionaryService() {
+        return dataDictionaryService;
+    }
+
+    @Override
+    public void setDataDictionaryService(DataDictionaryService dataDictionaryService) {
+        this.dataDictionaryService = dataDictionaryService;
+    }
+
+    /**
+     * @see org.kuali.kfs.pdp.batch.service.ExtractPaymentService#formatCheckNoteLines(java.lang.String)
+     *
+     * Long check stub note
+     */
+   @Override
+   public List<String> formatCheckNoteLines(String checkNote) {
+       List<String> formattedCheckNoteLines = new ArrayList<String>();
+
+       if (StringUtils.isBlank(checkNote)) {
+           return formattedCheckNoteLines;
+       }
+
+       String[] textLines = StringUtils.split(checkNote, BusinessObjectReportHelper.LINE_BREAK);
+       int maxLengthOfNoteLine = dataDictionaryService.getAttributeMaxLength(PaymentNoteText.class, "customerNoteText");
+       for (String textLine : textLines) {
+           String text = WordUtils.wrap(textLine, maxLengthOfNoteLine, BusinessObjectReportHelper.LINE_BREAK, true);
+           String[] wrappedTextLines = StringUtils.split(text, BusinessObjectReportHelper.LINE_BREAK);
+           for (String wrappedText : wrappedTextLines) {
```

Comment:
This could be replaced by a `Collections.addAll(...)`; another future opportunity, not enough to request another commit.